server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -352,7 +352,6 @@ public DocumentMapper merge(String type, CompressedXContent mappingSource, Merge
 
     private DocumentMapper newDocumentMapper(Mapping mapping, MergeReason reason, CompressedXContent mappingSource) {
         DocumentMapper newMapper = new DocumentMapper(documentParser, mapping, mappingSource);
-        newMapper.mapping().getRoot().fixRedundantIncludes();
```

Comment:
when would we end up performing this check with the new logic? It's performed earlier?

Reply:
Yes, it's now done when the Mapping is built in the first place.